_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,179 @@
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_step_2.png" alt="Release Push Image to Repository Step" title="Release Push Image to Repository Step" class="img img-responsive style-screengrab">
+
+### Save Version
+
+The image is safely stored in the repository, but we need to do one more step: save the version of our image. Once again we run some shell script to generate `image.version file: 
+
+```bash
+echo %build.number% > image.version
+```
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_step_3.png" alt="Release Save Version Step" title="Release Save Version Step" class="img img-responsive style-screengrab">
+
+## Deploy Application
+
+In the previous step we created an image which we can use now to deploy our application. We are going to create another build configuration: **Deploy**. This build will run a Docker container on **TeamCity** agent based on the image from **Release** phase.
+
+Our build configuration must contain three elements. The first one is a trigger. We will use again **Finish Build Trigger** with dependency to **Release** build configuration.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_trigger.png" alt="Deploy Trigger" title="Deploy Trigger" class="img img-responsive style-screengrab">
+
+The second element is a version of our image. We can obtain this information from **artifact** created by **Release** build. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_artifact.png" alt="Deploy Artifact" title="Deploy Artifact" class="img img-responsive style-screengrab">
+
+And the last one is step. The script for deployment is a little bit more complicated. 
+Before we deploy the new version of the application we need to check if it is not already deployed. If we already have running docker container with the previous version we need to stop and remove it. 
+Next we read current version from the **artifact** and create the new container based this version.  
+
+```
+if [ -n "$(docker ps --filter label=simple_application -q)" ]; then 
+   docker stop simple_application
+   docker rm simple_application
+fi;
+
+version_to_deploy=$(cat image.version)
+docker run -d -p 80:4567 \
+           --name simple_application \
+           --label simple_application \
+           registry.private/simple_application:${version_to_deploy}
+```
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_step.png" alt="Deploy Step" title="Deploy Step" class="img img-responsive style-screengrab">
+
+If we now run build of our application and we pass all tests **Release** and **Deploy** builds will start automatically and deploy our application.
+ 
+Continuous Deployment is ready.
+
+## It's done.
+
+Continuous deployment pipeline described in this post is of course simplified. Between our **Release** and **Deploy** steps we would like to do some 
```

Comment:
The continuous deployment pipeline described in this post is of course a simplified example.

Between our **Release** and **Deploy** steps we would likely to do some